machine.py

```diff
@@ -0,0 +1,121 @@
+from typing import Dict, Set
+
+from coin import Coin
+from costumer import Costumer
+from drink import Drink
+from resource import Resource
+
+
+def print_coins(coins: Dict[Coin, int]):
+    for coin in coins.keys():
+        print(f'{coin.name}: {coins[coin]}')
+
+
+class NoSuchDrinkError(Exception):
+    def __init__(self, msg: str):
```

Comment:
I might be rusty but I think the default implementation is to receive string msg. So a simple pass statement will suffice.